src/main/java/com/gtnewhorizon/gtnhlib/client/event/RenderTooltipEvent.java

```diff
@@ -0,0 +1,153 @@
+ * <p>
+ * <b>Note:</b> Use {@link ItemTooltipEvent} to modify the text being displayed.
+ * <p>
+ * This event is fired on the {@link MinecraftForge#EVENT_BUS} (client-side only).
+ * 
+ * @since 0.6.0
+ * @author glowredman
+ * @see ItemTooltipEvent
+ */
+@Cancelable
+public class RenderTooltipEvent extends Event {
+
+    public static final int ORIGINAL_BG_START = 0xF0100010;
+    public static final int ORIGINAL_BG_END = 0xF0100010;
+    public static final int ORIGINAL_BORDER_START = 0x505000FF;
+    public static final int ORIGINAL_BORDER_END = (ORIGINAL_BORDER_START & 0xFEFEFE) >> 1
+            | ORIGINAL_BORDER_START & 0xFF000000;
+
+    /**
+     * The ItemStack of which the tooltip is being displayed
+     */
+    @Nonnull
+    public final ItemStack itemStack;
+
+    /**
+     * The GUI in which the tooltip is being rendered
+     */
+    @Nonnull
+    public final GuiScreen gui;
+
+    /**
+     * The upper background color in ARGB format
+     * <p>
+     * Default value: {@code 0xF0100010}
+     */
+    public int backgroundStart;
+
+    /**
+     * The lower background color in ARGB format
+     * <p>
+     * Default value: {@code 0xF0100010}
+     */
+    public int backgroundEnd;
+
+    /**
+     * The upper border color in ARGB format
+     * <p>
+     * Default value: {@code 0x505000FF}
+     */
+    public int borderStart;
+
+    /**
+     * The lower border color in ARGB format
+     * <p>
+     * Default value: {@code 0x5028007F}
+     */
+    public int borderEnd;
+
+    /**
+     * The X coordinate of the mouse cursor
+     */
+    public int x;
+
+    /**
+     * The Y coordinate of the mouse cursor
+     */
+    public int y;
+
+    /**
+     * The FontRenderer used to render the tooltip's text
+     * <p>
+     * Default value: {@link GuiScreen#fontRendererObj} or {@link Item#getFontRenderer(ItemStack)} (if that method
+     * doesn't return {@code null})
+     */
+    @Nonnull
+    public FontRenderer font;
+
+    /**
+     * Optional hook to completely replace the rendering code. Will be used instead of the vanilla code if no
+     * {@code null}. The provided argument is the text to render.
+     * <p>
+     * <b>Note:</b> The usage may break compat with other mods, for example AppleCore!
+     * <p>
+     * Default value: {@code null}
+     * 
+     * @apiNote The following GL states will be disabled before calling this hook: {@link GL12#GL_RESCALE_NORMAL},
+     *          {@link GL11#GL_LIGHTING}, {@link GL11#GL_LIGHT0}, {@link GL11#GL_LIGHT1},
+     *          {@link GL11#GL_COLOR_MATERIAL}, {@link GL11#GL_DEPTH_TEST}. They will be re-enabled after the hook is
+     *          called. {@link Gui#zLevel GuiScreen.zLevel} and {@link RenderItem#zLevel GuiScreen.itemRender.zLevel}
+     *          must be set/reset by the hook! An <a
+     *          href=https://forge.gemwire.uk/wiki/Access_Transformers>AccessTransformer</a> may be needed for this.
+     */
+    @Nullable
```

Comment:
Very well documented.

It looks like setting the zLevel is still a potential pain point for custom renderers. I assume you're not setting it yourself in case a custom zLevel is desired, and you're not adding zLevel to the rendering event because someone could realistically want to use several zLevels?

If that's the case would it be reasonable to also pass the renderer a straightforward way to set zLevel since they will need it? Something like this maybe :
```java
public BiConsumer<List<String>, Consumer<Float>> alternativeRenderer;
//...
event.alternativeRenderer.accept(textLines, (Float zLevel) -> this.zLevel = zLevel);
```